src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4323,7 +4329,8 @@ LogicalResult ONNXScatterOp::inferShapes(
 
 LogicalResult ONNXScatterElementsOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
```

Comment:
Above code is correct, it might be wise to only set the output result type when `data` is ranked. Reason being that changes in the ops trigger a new wave of shape inference in all the consumer ops, I believe.

Reply:
Thanks for pointing out the problem.

Reply:
thanks